src/profile/Profile.tsx

```diff
@@ -2,49 +2,356 @@ import React, {useEffect} from "react";
+        }
+
+        //Button that you can use if you are logged to start and save edit.
+        if (loggedIn && !edit) { //If logged in but not in edit mode
+            setEditButton(<Button onClick={() => {
+                initialize(true);
+            }}>Edit</Button>);
+            //setDeleteProfileButtonBlock()//TODO set to empty <div/>
+
+        } else if (edit && loggedIn) {//If logged in and in edit mode
+            setEditButton(<Button onClick={saveChangesEdit}>Save changes</Button>)
+            //setDeleteProfileButtonBlock()//TODO set to deleteprofilebutton with function.
+        } else {
+            setEditButton(<div/>)
+        }
+    };
+
+    return (
+        <div>
+            {error}
+            {profileInformationBlock}
+            {passwordBlock}
+            {editButton}
+            {cancelEditButton}
+        </div>
+    )
 };
 
-export async function GetUserInformation(userId : string) : Promise<User>{
 
-    let options : RequestInit = {
-        method: "Get",
+/**
+ * maps redux state to props
+ * @param state of the redux
+ */
+const mapStateToProps = (state: any) => {
+    return {
+        auth: state.auth
+    };
+};
+
+export default withRouter(connect(mapStateToProps)(Profile));
+
+export async function GetUserInformation(userId: string): Promise<User> {
+    let options: RequestInit = {
+        method: "GET",
         headers: {
-            "Content-Type" : "application/json"
+            "Content-Type": "application/json"
         },
         mode: "cors",
         cache: "default"
     };
-    /**
-     * Url is now local TODO: change this when backend gets updated
-     */
-    let response : Response = await fetch(config.SERVICES.ACCOUNT_SERVICE_URL + "")
-    return initialUserState;
+
+    try {
+        let idRequest: string = "/" + userId;
+        let response: Response = await fetch(config.SERVICES.ACCOUNT_SERVICE_URL + idRequest, options);
+        let body = await response.text();
+        if (response.status === 200) {
+            return JSON.parse(body); //returns type User if backend is consistent.
+        } else {
+            return initialUserState;
+        }
+    } catch (Exception) {
+        console.log("PROFILE GET REQUEST EXCEPTION: " + Exception);
+        return initialUserState;
+    }
+}
+
+export async function UpdateUserInformation(user: User, token: string): Promise<boolean> {
+    console.log(JSON.stringify(user));
+    let options: RequestInit = {
+        method: "PUT",
+        body: JSON.stringify(user),
+        headers: {
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + token
+        },
+        mode: "cors",
+        cache: "default"
+    };
+
+
+        let response: Response = await fetch(config.SERVICES.ACCOUNT_SERVICE_URL + "/" + user.id, options);
+        if(response.status === 200){
+            return true;
+        }
+        else{
+            var text = await response.text();
```

Comment:
Veranderen naar const of let svp
```suggestion
            let text = await response.text();
```